packages/widget_toolkit_pin/lib/src/lib_pin_code_with_biometrics/blocs/pin_code_bloc.dart

```diff
@@ -52,6 +52,19 @@ class PinCodeBloc extends $PinCodeBloc {
     required this.autoPromptBiometric,
   }) {
     authenticated.connect().addTo(_compositeSubscription);
+    _$deleteDigitEvent
+        .map((_) => _pinCode.add(_pinCode.value.substring(
```

Comment:
- The map should do only only thing ... map one value to another. No side effects should be executed here
- Please use one listener for updating the `_pinCode` subjects
- Please use functional programming where the operators do one thing (at the current implementation the switchMap do some checks and implicitly assigns new value to the `_pinCode`)

You can achieve explicit, declarative, functional implementation such as

```
Rx.merge([
    _$deleteDigitEvent.map(/* get the latest value from the  _pinCode and emit a new state of it */),
    _$addDigitEvent.switchMap(/* get the latest value from the  _pinCode and emit a new state of it */),
]).listen(_pinCode.add).dispose...
.
```

Reply:
Refactored